pos_payment_terminal/static/src/js/devices.js

```diff
@@ -0,0 +1,114 @@
+                    return;
+                }
+            var paymentwidget = self.pos.chrome.screens.payment;
+            var drivers = status.newValue.drivers;
+            var order = self.pos.get_order();
+
+            // Check if the driver result is a terminal and then
+            // update payment lines with result
+            Object.keys(drivers).forEach(function(driver_name) {
+                if (drivers[driver_name].hasOwnProperty("is_terminal")) {
+                    self.update_payment_line(drivers[driver_name])
+                }
+            });
+            paymentwidget.order_changes();
+        },
+        update_payment_line: function(response){
+            var self = this;
+            var order = self.pos.get_order();
+            var paymentwidget = self.pos.chrome.screens.payment;
+            if ('transactions' in response){
+                for (var transaction_id in response.transactions){
+                    var transaction = response.transactions[transaction_id];
+                    if ('transaction_id' in transaction){
+                        var transaction_id = transaction['transaction_id'];
+                        var line = order.get_payment_line_by_transaction(transaction_id);
+                        if (line === undefined){
+                            continue;
+                        }
+                        if ('success' in transaction){
+                            line.terminal_transaction_success = transaction['success'];
+                        }
+                        if ('status' in transaction){
+                            line.terminal_transaction_status = transaction['status'];
+                        }
+                        if ('reference' in transaction){
+                            line.terminal_transaction_reference = transaction['reference'];
+                        }
+                        if (line){
+                            paymentwidget.transaction_changed(line);
+                        }
+                    }
+                }
+            }
+        },
+        update_transaction_data: function(line, data){
+            data.amount = line.get_amount();
+            data.payment_mode = line.cashregister.journal.payment_mode;
+        },
+        handle_terminal_response: function(line_id, response){
+            // This is intended to entrich payment line with
+            // useful values (status, transaction_id, ...)
+            // Note that transaction_id is mandatory to match response
+            if (response === undefined){
+                return;
+            }
+            var self = this;
+            var order = self.pos.get_order()
+            var line = order.get_paymentline(line_id)
+            var paymentwidget = self.pos.chrome.screens.payment;
+            if ('transaction_id' in response){
+                line.terminal_transaction_id = response['transaction_id']
+            }
+            if ('success' in response){
```

Comment:
this test seems useless: `line.terminal_success = response.success` should be sufficient.